core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java

```diff
@@ -197,6 +197,9 @@ private void updateBalancingState() {
 
   private void updateBalancingState(ConnectivityState state, RoundRobinPicker picker) {
     if (state != currentState || !picker.isEquivalentTo(currentPicker)) {
+      if (currentState == TRANSIENT_FAILURE && state != READY) {
+        return;
```

Comment:
I don't think we want to throw away picker changes.

Reply:
Yep. Ignore that change. Updated the PR according to offline discussion.